src/lib/merkle_ledger_tests/test_stubs.ml

```diff
@@ -109,12 +149,30 @@ end
 
 module Key = struct
   module T = struct
-    type t = string [@@deriving sexp, compare, hash, bin_io]
+    type t = Account.key [@@deriving sexp, bin_io, eq, compare, hash]
+
+    let gen = Account.key_gen
   end
 
-  let empty = ""
+  let empty = Account.empty.public_key
+
+  let to_string = Format.sprintf !"%{sexp: T.t}"
 
-  let to_string = Fn.id
+  let gen_keys num_keys =
+    (* TODO : the Quickcheck generator for Public_key.Compressed produces duplicates
+       as a workaround, we generate extra keys, remove duplicates, and take as many as needed
+       Issue #1078 notes the problem with the generators
+     *)
+    let num_to_gen = num_keys + (num_keys / 5) in
+    let more_than_enough_keys =
+      Quickcheck.random_value
+        (Quickcheck.Generator.list_with_length num_to_gen T.gen)
+    in
+    let unique_keys =
+      List.dedup_and_sort ~compare:T.compare more_than_enough_keys
+    in
+    assert (List.length unique_keys >= num_keys) ;
+    List.take unique_keys num_keys
```

Comment:
Wouldn't it be better to have a set and iteratively add keys to the set until you have `num_keys` unique keys? Then convert it to a list.

Reply:
Yes, will do.